include/sddf/network/queue.h

```diff
@@ -47,7 +47,7 @@ typedef struct net_queue_handle {
  *
  * @return number of buffers enqueued into a queue.
  */
-static inline uint32_t net_queue_size(net_queue_t *queue)
+static inline uint16_t net_queue_size(net_queue_t *queue)
 {
     return queue->tail - queue->head;
```

Comment:
We still need the `% size` if the size of the queues are less than 65535.

Reply:
Not necessary, as head and tail are unsigned.  The invariant maintained by the rest of the code is 0 <= tail - head <= SIZE.  head and tail just keep increasing without end chasing each other and wrapping around.

Reply:
Peter is right here, although we do need to ensure that our queue sizes are a power of two for other integrity constraints. I'll add that somewhere...